source/Outfit.h

```diff
@@ -44,6 +44,34 @@ class Outfit : public Weapon {
 	// These are all the possible category strings for outfits.
 	static const std::vector<std::string> CATEGORIES;
 
+
+public:
+	// A class that represents an outfir that makes other outfits.
+	class Factory {
+	public:
+		class Item {
+		public:
+			const Outfit *outfit = nullptr;
+			std::string name = "";
+			int count = 1;
+			bool asCargo = false;
+			bool isCommodity = false;
+		};
+
+	public:
+		std::vector<Item> input;
+		std::vector<Item> output;
+		double shield;
+		double hull;
+		double energy;
+		double fuel;
+		double heat;
+		int interval = 0;
```

Comment:
Please add a few comments on what interval, period and repeat are supposed to do.